x-pack/filebeat/input/http_endpoint/config.go

```diff
@@ -52,5 +56,11 @@ func (c *config) Validate() error {
 		}
 	}
 
+	if c.SecretHeader != "" || c.SecretValue != "" {
+		if c.SecretHeader == "" || c.SecretValue == "" {
```

Comment:
maybe more direct: `(c.SecretHeader != "" && c.SecretValue == nil) || (c.SecretHeader == "" && c.SecretValue == "")`

Reply:
I can make the conditions in the same format as you mentioned above, but it does not check the same thing as my own conditions.

The difference is that yours only check if `SecretHeader` is set, but not `SecretValue`, is that on purpose?